### PR TITLE
Gradle Arquillian Adapter Improvements

### PR DESCRIPTION
**Pull request by roguexz**
* Support for Gradle 5.0
* Handle the scenario where the IdeaModule marks a library dependency as "provided", because it is being provided by a project-dependency.

- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [ ] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java [hunk 1]
```diff
@@ -15,20 +15,29 @@
  */
 package org.wildfly.swarm.arquillian.adapter.gradle;
 
+import java.io.File;
```
Comment (roguexz) [inline arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java:18]: Please review the [original file](https://github.com/roguexz/thorntail/blob/master/arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java) to view the details of the approach better.

File: build-parent/pom.xml [hunk 2]
```diff
@@ -76,7 +76,7 @@
 
     <version.flyway>4.2.0</version.flyway>
 
-    <version.gradle>3.0</version.gradle>
+    <version.gradle>3.1</version.gradle>
```
Comment (roguexz) [inline build-parent/pom.xml:79]: By bumping to 3.1, I am able to make use of an API (not available in 3.0) that is available all the way till 5+ version. The original API that I was making use of was marked as deprecated in 3.0 and finally removed in 5.0.

**Comment by roguexz** [comment]
@Ladicek I can confirm that the changes work with Gradle 4.10.2 & 5.0 both from the command line as well as within the IDE. I have also verified that it works well with [composite builds](https://docs.gradle.org/current/userguide/composite_builds.html) too. Please review and provide your comments.

**Review by Ladicek — APPROVED** [review]
I don't know much about Gradle or our Gradle plugin anymore, but there's nothing obviously wrong here :-)

**Comment by roguexz** [comment]
Thank you @Ladicek. Once I am done with all the performance changes, I shall try convincing you to switch to Gradle :) .. Till then, I will keep hacking and using it.

**Comment by Ladicek** [comment]
I was playing with Gradle back in the 1.0-RCx days, but learned Maven a lot better since then. And Gradle changed massively as well. I need to try the Gradle Kotlin integration, that might be a very nice thing :-)

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**